hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -167,23 +167,27 @@ def get_metadata_objects_from_match(
         cls,
         signal_type: t.Type[SignalType],
         match: IndexMatch,
-    ) -> t.List[t.Union[PDQSignalMetadata]]:
+    ) -> t.List[t.Union[ThreatExchangeSignalMetadata]]:
         """
         See docstring of `write_signal_if_not_found` we will likely want to move
         this outside of Matcher. However while the MD5 expansion is still on going
         better to have it all in once place.
         Note: changes made here will have an effect on api.matches.get_match_for_hash
         """
         # TODO: Abstract out MD5SignalMetadata
-        if signal_type == PdqSignal:
+        if (
+            signal_type == PdqSignal
+            and match.metadata["source"]
+            == ThreatExchangeSignalMetadata.SIGNAL_SOURCE_SHORTCODE
+        ):
             return [
-                PDQSignalMetadata(
-                    str(match.metadata["id"]),
-                    privacy_group_id,
-                    datetime.datetime.now(),
-                    match.metadata["source"],
-                    match.metadata["hash"],
-                    match.metadata["tags"].get(privacy_group_id, []),
+                ThreatExchangeSignalMetadata(
+                    signal_id=str(match.metadata["id"]),
+                    privacy_group_id=privacy_group_id,
+                    updated_at=datetime.datetime.now(),
+                    signal_type=PdqSignal,
```

Comment:
nit: give the check it above on like 179 just pass in `signal_type` to make it even easier to generalize.